app-drawer-layout/app-drawer-layout.html

```diff
@@ -223,46 +216,31 @@
         }
       },
 
-      resetLayout: function() {
-        this.debounce('_resetLayout', function() {
-          var drawer = this.drawer;
-          var contentContainer = this.$.contentContainer;
+      _updateLayoutStates: function() {
+        var drawer = this.drawer;
+        var contentContainer = this.$.contentContainer;
 
-          if (this.narrow || !drawer) {
+        if (this.narrow || !drawer || !drawer.getWidth) {
+          contentContainer.style.marginLeft = '';
+          contentContainer.style.marginRight = '';
+        } else {
+          var drawerWidth = drawer.getWidth();
+          if (drawer.position == 'right') {
             contentContainer.style.marginLeft = '';
-            contentContainer.style.marginRight = '';
+            contentContainer.style.marginRight = drawerWidth + 'px';
           } else {
-            var drawerWidth = drawer.getWidth();
-            if (drawer.position == 'right') {
-              contentContainer.style.marginLeft = '';
-              contentContainer.style.marginRight = drawerWidth + 'px';
-            } else {
-              contentContainer.style.marginLeft = drawerWidth + 'px';
-              contentContainer.style.marginRight = '';
-            }
-          }
-        });
-      },
-
-      _resetDrawerState: function() {
```

Comment:
Open demo2.html, inspect `app-drawer-layout` and set `$0.openedWhenNarrow = true`, resize the window so it's narrow (the drawer should stay opened, but no longer persistent), close the drawer (e.g. click on the scrim), then resize the window again. Notice how the drawer becomes opened as soon as resize happens (expected: it stays closed)